bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/EpsonProjectorDevice.java

```diff
@@ -0,0 +1,609 @@
+            94, 99, 104, 109, 114, 120, 125, 130, 135, 141, 146, 151, 156, 161, 167, 172, 177, 182, 188, 193, 198, 203,
+            208, 214, 219, 224, 229, 235, 240, 245, 250 };
+
+    private static final int[] MAP20 = new int[] { 0, 12, 24, 36, 48, 60, 73, 85, 97, 109, 121, 134, 146, 158, 170, 182,
+            195, 207, 219, 231, 243 };
+
+    private static final int[] MAP18 = new int[] { 0, 13, 26, 40, 53, 67, 80, 94, 107, 121, 134, 148, 161, 175, 188,
+            202, 215, 229, 242 };
+
+    private static final int[] MAP_COLOR_TEMP = new int[] { 0, 25, 51, 76, 102, 128, 153, 179, 204, 230 };
+    private static final int[] MAP_FLESH_COLOR = new int[] { 0, 36, 73, 109, 146, 182, 219 };
+
+    private static final int DEFAULT_TIMEOUT = 5 * 1000;
+    private static final int POWER_ON_TIMEOUT = 100 * 1000;
+    private static final int POWER_OFF_TIMEOUT = 130 * 1000;
+
+    private static final String ON = "ON";
+    private static final String ERR = "ERR";
+
+    private Logger logger = LoggerFactory.getLogger(EpsonProjectorDevice.class);
+
+    private EpsonProjectorConnector connection;
+    private long lastLampQry = 0;
+    private int cachedLampHours = 0;
+    private boolean connected = false;
+    private boolean ready = false;
+
+    public EpsonProjectorDevice(SerialPortManager serialPortManager, String serialPort) {
+        connection = new EpsonProjectorSerialConnector(serialPortManager, serialPort);
+        ready = true;
+    }
+
+    public EpsonProjectorDevice(String ip, int port) {
+        connection = new EpsonProjectorTcpConnector(ip, port);
+        ready = true;
+    }
+
+    public EpsonProjectorDevice() {
+        connection = new EpsonProjectorDefaultConnector();
+        ready = false;
+    }
+
+    public boolean isReady() {
+        return ready;
+    }
+
+    private synchronized @Nullable String sendQuery(String query, int timeout)
+            throws EpsonProjectorCommandException, EpsonProjectorException {
+        logger.debug("Query: '{}'", query);
+        String response = connection.sendMessage(query, timeout);
+
+        if (response.length() == 0) {
+            throw new EpsonProjectorException("No response received");
+        }
+
+        response = response.replace("\r:", "");
+        logger.debug("Response: '{}'", response);
+
+        if (ERR.equals(response)) {
+            throw new EpsonProjectorCommandException("Error response received for command: " + query);
+        }
+
+        if ("PWR OFF".equals(query) && ":".equals(response)) {
+            // When PWR OFF command is sent, next command can be sent 10 seconds after the colon is received
+            logger.debug("Refusing further commands for 10 seconds to power OFF completion");
+            ready = false;
+            scheduler.scheduleWithFixedDelay(() -> {
+                ready = true;
+            }, 0, 10000, TimeUnit.MILLISECONDS);
```

Comment:
This will schedule a periodic task. I think your intention is to schedule an one-shot?

Reply:
It needs to poll the projector continuously to detect changes in state from using the physical remote control.

Reply:
Then, you need to cancel the task when the Thing handler's `dispose()` is called.

Reply:
My bad,  I was looking at the wrong scheduler call.